File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/scheduler/SchedulerHttpService.java
package com.sos.scheduler.engine.kernel.scheduler;

import com.google.inject.ImplementedBy;
import com.sos.scheduler.engine.kernel.Scheduler;
import com.sos.scheduler.engine.kernel.cppproxy.HttpResponseC;
import com.sos.scheduler.engine.kernel.http.SchedulerHttpRequest;
import com.sos.scheduler.engine.kernel.http.SchedulerHttpResponse;

@ImplementedBy(Scheduler.class)
public interface SchedulerHttpService {
    /** @return {@link HttpResponseC#Release()} ()} MUSS aufgerufen werden! */
    HttpResponseC executeHttpRequest(SchedulerHttpRequest request, SchedulerHttpResponse response);
}

File: opensource/kernel/src/main/scala/com/sos/scheduler/engine/kernel/scheduler/SchedulerHttpService.scala
package com.sos.scheduler.engine.kernel.scheduler

import com.sos.scheduler.engine.kernel.http.{SchedulerHttpResponse, SchedulerHttpRequest}
import com.sos.scheduler.engine.kernel.cppproxy.{SpoolerC, HttpResponseC}
import javax.inject.{Inject, Singleton}

@Singleton
class SchedulerHttpService @Inject private(cppProxy: SpoolerC) {

  /** @return { @link HttpResponseC#close()} MUSS aufgerufen werden! */
  def executeHttpRequest (request: SchedulerHttpRequest, response: SchedulerHttpResponse): HttpResponseC =
    cppProxy.java_execute_http (request, response)
}
